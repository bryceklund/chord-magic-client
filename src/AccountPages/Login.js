import React, { Component } from 'react';
import Nav from '../Nav/Nav.js';
import { Link } from 'react-router-dom';
import './Account.css';

class Login extends Component {
    constructor(props) {
        super(props)
        this.state = {
            username: null,
            password: null
        }
    }

    setUser = (username) => {
        this.setState({
            username
        })
    }

    setPass = (password) => {
        this.setState({
            password
        })
    }

    render() {
        return (
            <body>
                <section className='login'>
                    <form className='login_form'>
                        <fieldset>
                        <legend><h3>Login</h3></legend>
                        <label htmlFor='username'>Username:</label>
                        <input type='text' id='username' onChange={(e) => this.setUser(e.target.value)} />
                        <label htmlFor='password'>Password:</label>
                        <input type='password' id='password' onChange={(e) => this.setPass(e.target.value)} />
                        <button type="submit" onClick={() => this.props.signIn(this.state.username, this.state.password)}>Submit</button>
                        </fieldset>
                    </form>
                </section>
            </body>
        );
    }
}

export default Login;