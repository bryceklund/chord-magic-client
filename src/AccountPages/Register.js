import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Nav from '../Nav/Nav.js';
import './Account.css';

class Register extends Component {
    render() {
        return (
            <body>
                <Nav />
                <section className='login'>
                    <form className='login_form'>
                        <fieldset>
                        <legend><h3>Register</h3></legend>
                        <label htmlFor='username'>Desired Username:</label>
                        <input type='text' id='username' />
                        <label htmlFor='password'>Password:</label>
                        <input type='password' id='password' />
                        <label htmlFor='password'>Repeat Password:</label>
                        <input type='password' id='password' />
                        <button type="submit">Submit</button>
                        </fieldset>
                    </form>
                </section>
            </body>
        );
    }
}

export default Register;