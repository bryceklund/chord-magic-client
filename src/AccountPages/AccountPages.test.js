import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Login from './Login'
import Register from './Register'

describe(`Account Pages`, () => {
    it(`login renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter>
                            <Login />
                        </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })
    it(`login matches its snapshot`, () => {
        const tree = renderer.create(<BrowserRouter>
                                        <Login />
                                    </BrowserRouter>)
                                    .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it(`register renders without crashing`, () => {
        const div = document.createElement('html') //body cannot be a child element of ...anything?
        ReactDOM.render(<BrowserRouter>
                            <Register />
                        </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })
    it(`register matches its snapshot`, () => {
        const tree = renderer.create(<BrowserRouter>
                                        <Register />
                                    </BrowserRouter>)
                                    .toJSON()
        expect(tree).toMatchSnapshot()
    })
})