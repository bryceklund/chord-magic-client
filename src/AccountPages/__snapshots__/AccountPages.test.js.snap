// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Pages login matches its snapshot 1`] = `
<section
  className="login"
>
  <form
    className="login_form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        <h3>
          Login
        </h3>
      </legend>
      <label
        htmlFor="username"
      >
        Username:
      </label>
      <input
        id="username"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        id="password"
        onChange={[Function]}
        type="password"
      />
      <button
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </fieldset>
  </form>
  <p
    className="error hidden"
    id="error"
  >
    
  </p>
</section>
`;

exports[`Account Pages register matches its snapshot 1`] = `
<body>
  <nav>
    <h1>
      <a
        href="/editor"
        onClick={[Function]}
      >
        Chord Magic
      </a>
    </h1>
    <a
      href="/editor"
      onClick={[Function]}
    >
      <img
        alt="chord magic logo"
        src=""
      />
    </a>
    <div
      className="login_links"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      Â 
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </nav>
  <section
    className="login"
  >
    <form
      className="login_form"
      onSubmit={[Function]}
    >
      <fieldset>
        <legend>
          <h3>
            Register
          </h3>
        </legend>
        <label
          htmlFor="username"
        >
          Desired Username:
        </label>
        <input
          id="username"
          onChange={[Function]}
          type="text"
        />
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <input
          id="password_one"
          onChange={[Function]}
          type="password"
        />
        <label
          htmlFor="password"
        >
          Repeat Password:
        </label>
        <input
          id="password"
          onChange={[Function]}
          type="password"
        />
        <p
          className="error "
        />
        <button
          disabled={true}
          type="submit"
        >
          Submit
        </button>
      </fieldset>
    </form>
  </section>
</body>
`;
